packages/models-library/src/models_library/basic_types.py

```diff
@@ -102,6 +123,11 @@ def __get_pydantic_core_schema__(cls, _source_type, _handler):
 
 
 class IDStr(ConstrainedStr):
+    """Non-empty bounded string used as identifier
+
+    DEPRECATED: Use instead Annotated[str, StringConstraints(strip_whitespace=True, min_length=1, max_length=100)]
```

Comment:
Not sure I get the subtle difference between this and the new query strings you introduce. Using the constraints like suggested appears less straightforward to me.

Reply:
Also this class brings with tit the concatenate staticmethod